File: Data Structures & Algorithms/counting-bits/submission-1.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> ans = {0,1};
        if (n==0)
            return {0};
        if(n==1){
            return {0,1};
        }
        int check = 2;
        for(int i =2;i<=n;i++){
            if(i==2*check){
                check = i;
            }
            ans.push_back(ans[i-check]+1);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        
        ans = 0
        def req(root,k):
            nonlocal ans
            if root==None:
                return
            
            req(root.left,k)

            if k[0]==1:
                ans = root.val
            k[0]-=1
            req(root.right,k)
            return
        
        req(root,[k])
        return ans

